docker/package/tezos_setup_wizard.py

```diff
@@ -77,7 +77,7 @@
 def fetch_snapshot(url):
     print("Downloading the snapshot from", url)
     filename = TMP_SNAPSHOT_LOCATION
-    urllib.request.urlretrieve(url, filename, progressbar_hook)
+    proc_call(f"wget --show-progress --continue -O {filename} {url}")
     print()
     return filename
```

Comment:
Have you considered what will happen to this in all the cases too?

For example, I didn't test this, but here's some questions worth answering:
1. is it possible to resume a download from a previous `tezos-setup` run?
2. what happens when there is still an incomplete snapshot on disk and this code is run?
3. what happens if the user changes their URL or decides to switch from using the metadata to a URL?
4. what if the latest snapshot from the metadata is pointing at a different one?

Reply:
1. Yes, AFAIU [download resume works on server-side](https://stackoverflow.com/questions/12434306/in-general-how-does-resuming-a-download-work), so it doesn't matter how and why download stops and when and how we resume it. It just will check the state of file and will try to do as little of work as possible.
2. If you mean it's totally different snapshot, it will be wiped in favor of file we want to download.
3 and 4 questions is irrelevant IIUC, because this code works already knowing the url of snapshot we want to download, disregarding the source

Reply:
> it doesn't matter how and why download stops and when and how we resume it

It does matter if the partial file on disk is not the same file we are trying to download anymore.

As mentioned in your link, the request to use a specific range is how download resuming works, but that assumes that we are indeed re-downloading the same file. If we "resume" the wrong download then we'll have a file in the end that's half our first attempt and half the second one.

I asked:
> is it possible to resume a download from a previous tezos-setup run?

both as a negative scenario and a positive one. 
IOW the question is: is it possible to incorrectly resume a download that was interrupted from a previous run?
Or will the previous snapshot be deleted in favor of a new download?
And is it possible to correctly resume a download that was interrupted in a previous run?

> If you mean it's totally different snapshot, it will be wiped in favor of file we want to download

Great, that's what should happen :+1: 

How do we decide if the existing incomplete file is from a "totally different snapshot" tho?
Can we also distinguish between different snapshots that are served on the same URL (at different times, e.g. if the URL given by the user is for the "latest" snapshot)?

P.s. Question 3 and 4 are basically reiteration of the same ones.
I wanted to point out that we need to handle all the different cases in which we are trying to resume a download.

Reply:
> IOW the question is: is it possible to incorrectly resume a download that was interrupted from a previous run?

Seems like, it is. Thanks for noticing it.